### ISSUE TITLE
[service] Installer packages can't be used in older enviroments 

### ISSUE THREAD

**Issue by Croydon** [conan-io/conan-center-index#213]
For some reason installers build by CCI seems to use the new CXX11 ABI which then doesn't run in GCC 4.9 containers. Am I missing something here or is this effectively a bug? 🤔

https://travis-ci.com/bincrafters/conan-sdl2/jobs/247592499#L1801

// @ericLemanissier @SSE4

**Comment by ericLemanissier** [comment]
another occurrence of this problem: https://travis-ci.com/bincrafters/conan-xkbcommon/jobs/258981515#L1062
This happens probably because [ninja recipe deletes `self.info.settings.compiler`](https://github.com/conan-io/conan-center-index/blob/master/recipes/ninja/all/conanfile.py#L61). Maybe it it could be fixed by deleting all subkeys of `self.settings.compiler` except `libcxx` ?

**Comment by danimtb** [comment]
I believe that's not going to solve the problem, as `compiler.libcxx` does not model `glibc` but the C++ standard library

**Comment by ericLemanissier** [comment]
the missing symbol is `_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE7reserveEm` which is demangled as `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::reserve(unsigned long)` so it definitely looks like a c++ standard library issue to me.  Platforms with gcc 4.9 cannot have C++11 version of the abi, because it was introduced in gcc 5.1

**Comment by uilianries** [comment]
The installer packages should use the centos6 docker image... We need to check it.

**Comment by uilianries** [comment]
I can't access ninja installer CI log, it's so old. Anyway, I think we are using conanio/gcc5

**Comment by ericLemanissier** [comment]
another possibility is to compile ninja with `-static-libstdc++` so that it does not depend on dynamic libraries at all. Would such a change be accepted ? Other installer packages using std library would still have the problem.

**Comment by SSE4** [comment]
unfortunately, `-static-libstdc++` isn't enough :( still, glibc problem arises after that, which is not that easy to deal.

**Comment by ericLemanissier** [comment]
so all installer packages should be built on the platform with oldest glibc right ? so CCI should use gcc4.9 image for these ?

**Comment by SSE4** [comment]
yeah, I believe so, it's the most simple solution. however, CCI doesn't have a concept of `installer` packages, and there is no way to specify such information that package must be built in a special way.

**Comment by SSE4** [comment]
@danimtb @uilianries  we need to definitely make some action on this, as it blocks several recipes (e.g. `fribidi`).

**Comment by Croydon** [comment]
If the CI system would be open sourced... 😉  

Random wild guess: Maybe GCC 5 is the first compiler the CI compiles for and therefore uses it for installer packages? Maybe it would be enough to put GCC 4.9 first?

**Comment by ericLemanissier** [comment]
another PR blocked because of this: https://github.com/conan-io/conan-center-index/pull/1013

**Comment by ericLemanissier** [comment]
So, we're in the absurd situation where this issue which happens only on linux prevents from releasing a package (pkgconf) which is actually only needed on windows (because pkg-config is available on linux at the system level), and it ends up blocking all packages built with meson. 
Could we consider a pragmatic solution, for example what about modifying the ninja recipe like this:
```
if os.getenv("C3I_ENV_VAR"):
    if self.settings.compiler == "gcc" and self.settings.compiler.version != "4.9":
        raise ConanInvalidConfiguration("distro is too new")
```

**Comment by ericLemanissier** [comment]
another workaround solution: drop gcc 4.9 on CCI supported platforms (that has the another benefit : freeing a slot for gcc10)

**Comment by danimtb** [comment]
I'd consider this issue closed with the documentation in #3328
